### PR TITLE
Intermittent protection 0.17.x

### PR DESCRIPTION
**Pull request by bbondy**

### CODE REVIEW

File: test/knownFailures.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Known intermittent failures
+
+This file is parsed when running automated tests on our CI.
+
+Known failures are shown in logs, but status will be green unless it is an uknown failure.
```
Comment (luixxiul) [inline test/knownFailures.md]: nit: unknown

File: test/knownFailures.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+## bravery-components
+
+- Autoplay test "after each" hook for "allow autoplay and remember"
+- Bravery Panel Adblock stats iframe tests detects blocked elements in iframe
+- Bravery Panel Adblock stats iframe tests detects blocked elements in iframe in private tab
+- Bravery Panel Adblock stats without iframe tests block device enumeration
+- Bravery Panel Adblock stats without iframe tests blocks custom adblock resources in private tab
+- Bravery Panel Adblock stats without iframe tests blocks scripts in a private tab
+- Bravery Panel Adblock stats without iframe tests blocks scripts in a regular tab
```
Comment (diracdeltas) [inline test/knownFailures.md]: are the block script exceptions necessary? i haven't seen them fail much

File: test/knownFailures.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+- Bravery Panel Adblock stats without iframe tests blocks websocket tracking
+- Bravery Panel Adblock stats without iframe tests detects adblock resources in private tab
+- Bravery Panel Adblock stats without iframe tests downloads and detects regional adblock resources
+- Bravery Panel Adblock stats without iframe tests downloads and detects regional adblock resources in private tab
+- Bravery Panel Tracking Protection stats detects blocked elements
+- Bravery Panel Tracking Protection stats detects blocked elements in private tab
+- Clear Browsing Panel with closedFrames clears closedFrames
+- Clear Browsing Panel with saved state values saves the history switch state
+- noscript info can allow scripts on a file:// URL without allowing all scripts
+- noscript info can selectively allow scripts
```
Comment (diracdeltas) [inline test/knownFailures.md]: noscript tests should be kept if possible

File: test/knownFailures.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+## contents
+
+- content loading does not support battery status API
+
+## misc-components
+
+- Syncing and clearing data prevents it from syncing history
+- Syncing and clearing data prevents it from syncing site settings
+- Syncing bookmarks from an existing profile update bookmark, moving it into the folder
+- Syncing bookmarks update bookmark, moving it into the folder
```
Comment (diracdeltas) [inline test/knownFailures.md]: the syncing bookmarks tests should not be exceptions

**Comment by bsclifton** [comment]
Closing as 0.17.x is no longer relevant

Did we ever come to a consensus about using the proposed code? I think this was an improvement for sure- wouldn't want to lose the work done

**Pull request closed without merging**